README.md

```diff
@@ -170,54 +168,75 @@ sensor:
-      - { phase_id: phase_a, input:  "5", power: { name:  "Circuit 5 Power", id:  cir5, filters: [ *moving_avg, *pos, multiply: 2 ] } }
-      - { phase_id: phase_a, input:  "6", power: { name:  "Circuit 6 Power", id:  cir6, filters: [ *moving_avg, *pos, multiply: 2 ] } }
-      - { phase_id: phase_a, input:  "7", power: { name:  "Circuit 7 Power", id:  cir7, filters: [ *moving_avg, *pos, multiply: 2 ] } }
-      - { phase_id: phase_b, input:  "8", power: { name:  "Circuit 8 Power", id:  cir8, filters: [ *moving_avg, *pos ] } }
-      - { phase_id: phase_b, input:  "9", power: { name:  "Circuit 9 Power", id:  cir9, filters: [ *moving_avg, *pos ] } }
-      - { phase_id: phase_b, input: "10", power: { name: "Circuit 10 Power", id: cir10, filters: [ *moving_avg, *pos ] } }
-      - { phase_id: phase_a, input: "11", power: { name: "Circuit 11 Power", id: cir11, filters: [ *moving_avg, *pos, multiply: 2 ] } }
-      - { phase_id: phase_a, input: "12", power: { name: "Circuit 12 Power", id: cir12, filters: [ *moving_avg, *pos, multiply: 2 ] } }
-      - { phase_id: phase_a, input: "13", power: { name: "Circuit 13 Power", id: cir13, filters: [ *moving_avg, *pos ] } }
-      - { phase_id: phase_a, input: "14", power: { name: "Circuit 14 Power", id: cir14, filters: [ *moving_avg, *pos ] } }
-      - { phase_id: phase_b, input: "15", power: { name: "Circuit 15 Power", id: cir15, filters: [ *moving_avg, *pos ] } }
-      - { phase_id: phase_a, input: "16", power: { name: "Circuit 16 Power", id: cir16, filters: [ *moving_avg, *pos ] } }
+      - { phase_id: phase_a, input:  "1", power: { id:  cir1, filters: [ *pos ] } }
+      - { phase_id: phase_b, input:  "2", power: { id:  cir2, filters: [ *pos ] } }
+      - { phase_id: phase_a, input:  "3", power: { id:  cir3, filters: [ *pos ] } }
+      - { phase_id: phase_a, input:  "4", power: { id:  cir4, filters: [ *pos ] } }
+      - { phase_id: phase_a, input:  "5", power: { id:  cir5, filters: [ *pos, multiply: 2 ] } }
+      - { phase_id: phase_a, input:  "6", power: { id:  cir6, filters: [ *pos, multiply: 2 ] } }
+      - { phase_id: phase_a, input:  "7", power: { id:  cir7, filters: [ *pos, multiply: 2 ] } }
+      - { phase_id: phase_b, input:  "8", power: { id:  cir8, filters: [ *pos ] } }
+      - { phase_id: phase_b, input:  "9", power: { id:  cir9, filters: [ *pos ] } }
+      - { phase_id: phase_b, input: "10", power: { id: cir10, filters: [ *pos ] } }
+      - { phase_id: phase_a, input: "11", power: { id: cir11, filters: [ *pos, multiply: 2 ] } }
+      - { phase_id: phase_a, input: "12", power: { id: cir12, filters: [ *pos, multiply: 2 ] } }
+      - { phase_id: phase_a, input: "13", power: { id: cir13, filters: [ *pos ] } }
+      - { phase_id: phase_a, input: "14", power: { id: cir14, filters: [ *pos ] } }
+      - { phase_id: phase_b, input: "15", power: { id: cir15, filters: [ *pos ] } }
+      - { phase_id: phase_a, input: "16", power: { id: cir16, filters: [ *pos ] } }
+    on_update:
```

Comment:
This doesn't seem to work like the rest of esphome components. Isn't this what `on_value` is for?

https://esphome.io/components/sensor/index.html#on-value

Reply:
It has a very similar purpose. The ``on_value`` trigger executes when a specific sensor publishes a new value. With this component, we update more than one sensor each update; e.g., each CT clamp's power, voltage, etc... When we compute the total power and balance power sensors, we want to execute the template calculation exactly once after all the individual power sensors have updated from the new measurements. We could potentially use ``on_value`` trigger for the sensor that is last updated, but it isn't obvious which one that is unless we dive into the source code. The ``on_update`` trigger as implemented here is guaranteed to execute after all sensors have updated in the component.

When I asked about a similar issue on Discord recently, the recommendation was to add a new trigger (and I have done so in my [statistics component PR](https://github.com/esphome/esphome/pull/5103) component PR). That hasn't been merged yet, so it is possible I will get a different suggestion on review. If so, I will update this accordingly.

Reply:
Oh, I see! The issue is that you want to update all these values simultaneously. That makes sense. I'm curious to see how this works out. Thanks for explaining.

Reply:
Exactly, no worries! I struggle with clarity and conciseness sometimes, and I think your word simultaneously goes a long way to describe what the trigger does. I will add that to my description as well, thanks.